### PR TITLE
[TASK] Rewrite Index Inspector using Extbase/Fluid

### LINKED ISSUE #34: [TASK] Rewrite Index Inspector using Extbase/Fluid
Let's remove ExtJs from the Index Inspector backend module.

Possible replacement: http://www.jquery-bootgrid.com/

### LINKED ISSUE #948: [TASK] Removed usage of Ext.grid.RowExpander.js - deprecated
The usage of Ext.grid.RowExpander.js is deprecated with [TYPO3 8.x](https://docs.typo3.org/typo3cms/extensions/core/8-dev/Changelog/8.4/Breaking-77783-RemoveUnusedExtJSLibraries.html?highlight=rowexpander) - for the moment the version from TYPO3 7.6.x has been added as a temporary solution as part of #901

### PR DESCRIPTION
**Pull request by dkd-kaehm**
Using Extbase controller for „third level module“.

This closes #34, closes #948

### CODE REVIEW

**Comment by thomashohn** [comment]
Does it also fix #948 directly? Else please consider doing that too after this PR has been merged :-)

**Comment by timohund** [comment]
@dkd-kaehm thanks for the pull request. I have the following comments:

* There are several prs-2 violations. Can you please run php-cs-fixer?
* What about tests for the new ViewHelpers and the other code?
* @thomashohn pointed out a reasonable comment, we need to check that too

File: Classes/Domain/Repository/ApacheSolrDocumentRepository.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
+<?php
+namespace ApacheSolrForTypo3\Solr\Domain\Repository;
```
Comment (timohund) [inline Classes/Domain/Repository/ApacheSolrDocumentRepository.php]: I would propose to add a unit and integration test to make sure that we can retrieve the documents form solr in the future.

File: Classes/ViewHelpers/IsBoostedDocumentFieldViewHelper.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+
+use TYPO3\CMS\Fluid\Core\ViewHelper\AbstractConditionViewHelper;
+
+/**
+ * Condition for checking if type is a string.
+ */
+class IsBoostedDocumentFieldViewHelper extends AbstractConditionViewHelper
```
Comment (timohund) [inline Classes/ViewHelpers/IsBoostedDocumentFieldViewHelper.php]: I would propose to add a unit test for this view helper

File: Classes/ViewHelpers/IsStringViewHelper.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
```
Comment (timohund) [inline Classes/ViewHelpers/IsStringViewHelper.php]: I would propose to add a unit test for the ViewHelper

File: ext_tables.php [hunk 4]
```diff
@@ -52,14 +52,24 @@
...
+            'Administration' => 'index,setSite,setCore,noSiteAvailable',
+            'Backend\\Web\\Info\\ApacheSolrDocument' => 'index'
         ],
         [
             'access' => 'admin',
             'icon' => 'EXT:solr/Resources/Public/Images/Icons/ModuleAdministration.svg',
             'labels' => 'LLL:EXT:' . $_EXTKEY . '/Resources/Private/Language/locallang.xlf',
         ]
     );
+//    \TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerModule(
```
Comment (timohund) [inline ext_tables.php]: We should remove this

File: Classes/Backend/IndexInspector/BackendIndexInspectorModuleBootstrap.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
+<?php
+namespace ApacheSolrForTypo3\Solr\Backend\IndexInspector;
+
+/***************************************************************
+ *  Copyright notice
+ *
+ *  (c) 2012-2015 Ingo Renner <ingo@typo3.org>
```
Comment (irnnr) [inline Classes/Backend/IndexInspector/BackendIndexInspectorModuleBootstrap.php]: unless everything is just copied from somewhere else

File: ext_tables.php (hunk 4 above)
Comment (irnnr) [inline ext_tables.php]: yeah, no dead code

**Pull request merged**